src/scripts/main.js

```diff
@@ -1 +1,105 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let isResolved = false;
+
+  const clickHandler = () => {
+    if (!isResolved) {
+      isResolved = true;
+      clearTimeout(timer);
+      document.removeEventListener('click', clickHandler);
+
+      resolve('First promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  const timer = setTimeout(() => {
+    if (!isResolved) {
+      isResolved = true;
+      document.removeEventListener('click', clickHandler);
+
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
```

Comment:
It's a good practice to remove event listeners when they are no longer needed to avoid potential memory leaks. However, the 'clickHandler' is being removed inside the handler itself, which is correct, but it's not being removed if the promise is rejected by the timer. You should also remove the 'click' event listener in the timer's callback to ensure it's cleaned up in all cases.